zilencer/models.py

```diff
@@ -84,6 +84,35 @@ def __str__(self) -> str:
         return f"{self.server!r} {self.user_id}"
 
 
+class RemoteRealm(models.Model):
+    """
+    Each object corresponds to a single remote Realm that is using the
+    Mobile Push Notifications Service via `manage.py register_server`.
+    """
+
+    server = models.ForeignKey(RemoteZulipServer, on_delete=models.CASCADE)
+
+    # The unique UUID and secret for this realm.
+    uuid = models.UUIDField(unique=True)
+    uuid_owner_secret = models.TextField()
+
+    subdomain = models.TextField()
+
+    # The fields below are analogical to RemoteZulipServer fields.
+
+    last_updated = models.DateTimeField("last updated", auto_now=True)
+
+    # Whether the realm registration has been deactivated.
+    deactivated = models.BooleanField(default=False)
+
+    # Plan types for self-hosted customers
+    PLAN_TYPE_SELF_HOSTED = 1
+    PLAN_TYPE_STANDARD = 102
+
+    # The current billing plan for the remote server, similar to Realm.plan_type.
+    plan_type = models.PositiveSmallIntegerField(default=PLAN_TYPE_SELF_HOSTED, db_index=True)
```

Comment:
Feedback on what we want to adjust on this model?

- I changed the model fields compared to the original proposal which copied from `RemoteZulipServer` a bunch, since things like `hostname` or `contact_email` don’t make sense / duplicate info obtainable from `.server`.

- I’m also not sure if we want to store in the database all the fields submitted to us when posting analytics (e.g. `realm_uri` or `realm.host` since again - they kinda duplicate data? I just have the `subdomain` field here on this model, and the rest should be inferrable from that + the server’s `hostname`)

Reply:
Because `REALM_HOSTS` exists, I think we probably do want `realm.host` rather than `subdomain`.

Hmm. We definitely will want some form of separate `contact_email`, because the "in the web app UI" authentication flow and billing relationship for a realm will be attached to the realm and its administrators, not as a server administrators, who might be a different department/company. 

(This value should be populated as part of logging in).

But let's plan to add that later; I'm not sure exactly how it should work and it might prefer a separate table.

Reply:
Makes sense, I added `realm_host` instead of `subdomain`